src/api/ApiService.js

```diff
@@ -0,0 +1,302 @@
+import {
```

Comment:
I think that generally some of the logic can get some improvements, but that's anyway a huge pr (and it shouldn't be :P)
The last thing that you should add is:
-fix the conflicts between lint and prettier
-add hook to handle request with loading (it can for example accept the method from service and array of arguments ~ return data and loading state)

Then we can go ahead and and merge that I think, it wouldn't block any other task, and also then we can refactor some logic without impact for the "UI" layer

Reply:
aaaaa, you resolved the thread without answear and requested rereview but I don't see the hook anywhere

Reply:
The hook is at `hooks/useUser.jsx`. We don't need the `isLoading` property, because that makes no sense. If data is `null` it means that it is loading. If the user is not logged in, then the `ProtectedRoute` will deal with it and won't even render the component. So we have only two states left: loading and finished. If data is null it means that it is loading, if it is not - it has finished loading. Am I missing something or is it correct?

Reply:
That'a not how loading state works,

Null data =\= loading

If api would return an empty data, that means that promise is resolved, but with case that you are talking about we would have infinite loading

Loading is separate state which is based only on state of the promise and can't be based on state of the data

Using the data as an indicator for loading state is really bad

Reply:
Ok then... I will try to do this with `react-query` just as we discussed and agreed earlier with the team.

Reply:
There is a package that connects `react-query` and `firebase`. It is called simply `react-query-firebase`. Am I allowed to use it?

Reply:
Never used that, if you find that useFull and trustfull then go ahead